### PR TITLE
Thread-safe conversion for HLT modules in CommonTools/RecoAlgos

### PR DESCRIPTION
**Pull request by cvuosalo**
HLT modules LargestEtCaloJetSelector, LargestEtPFJetSelector, and RecoTrackRefSelector in CommonTools/RecoAlgos migrated to be thread safe. This PR was created in response to a request from Andrea Bocci: https://hypernews.cern.ch/HyperNews/CMS/get/recoDevelopment/1353.html

This change should have no effect on monitored quantities.

### CODE REVIEW

**Comment by cvuosalo** [comment]
@cmsbuild please test

File: CommonTools/RecoAlgos/interface/TrackSelector.h [hunk 1]
```diff
@@ -130,7 +132,7 @@ namespace helper {
 
 
   //----------------------------------------------------------------------
-  class TrackSelectorBase : public edm::EDFilter {
+  class TrackSelectorBase : public edm::stream::EDFilter<> {
   public:
     TrackSelectorBase( const edm::ParameterSet & cfg ) {
       std::string alias( cfg.getParameter<std::string>( "@module_label" ) );
```
Comment (slava77) [inline CommonTools/RecoAlgos/interface/TrackSelector.h]: the template instantiation for the StoreManagerTrait should be updated as well 
see its parameters at 
https://github.com/cms-sw/cmssw/blob/CMSSW_7_5_X/CommonTools/UtilAlgos/interface/StoreManagerTrait.h#L99

File: CommonTools/RecoAlgos/plugins/LargestEtPFJetSelector.cc [hunk 2]
```diff
@@ -6,7 +6,7 @@
  *
  */
 #include "FWCore/Framework/interface/MakerMacros.h"
-#include "CommonTools/UtilAlgos/interface/ObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/ObjectSelectorStream.h"
```
Comment (slava77) [inline CommonTools/RecoAlgos/plugins/LargestEtPFJetSelector.cc]: this is not enough, the ``` typedef ObjectSelector ``` should be changed as well

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/9081	 22224d9
- changes in the code are as expected
- jenkins tests run OK and comparisons with baseline show no difference, as expected

**Comment by davidlange6** [comment]
+1

**Pull request merged**